### PR TITLE
Main menu - New options button

### PR DESCRIPTION
**Pull request by Pacasasgar**
This PR contains the following changes:
-Now pause menu use audio_manager to play audios.
-New placeholder 'options' button in the MainMenu.
-Inside 'options' panel there is a new UI that allows to change volume.
-Volume logic is implemented in that new panel.
PD: Resolution has been hided for now because it's not implemented, just the UI.

options_improved_placeholder
options_panel_without_res

### CODE REVIEW

**Comment by tonifj** [comment]
I thought we also would add new icons for the arrows, at the options menu!

**Comment by Pacasasgar** [comment]
> I thought we also would add new icons for the arrows, at the options menu!

Yes yes, all of this is placeholder. The art part will be done during the tuning phase.

File: Scripts/src/Script/OptionsMenuController.cpp [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+#include "Module/ModuleTime.h"
+
+#include "AudioManager.h"
+#include "InputManager.h"
+#include "MainMenuController.h"
+#include "LevelSelectionMenuController.h"
+#include "UIMainMenuInputController.h"
+#include "InputManager.h"
+#include "WorldManager.h"
+
```
Comment (RickyVimon) [inline Scripts/src/Script/OptionsMenuController.cpp:23]: CS alphabetical order

File: Scripts/src/Script/OptionsMenuController.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	void MoveCursor();
+	void ManageVolumeUI();
+
+	void Open();
+	void Close();
+
+private:
+	GameObject* options_panel = nullptr;
+	GameObject* main_menu_panel = nullptr;
+
```
Comment (RickyVimon) [inline Scripts/src/Script/OptionsMenuController.h:42]: Delete all the pointers on the destructor

File: Scripts/src/Script/OptionsMenuController.h [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	GameObject* audio_controller = nullptr;
+	ComponentAudioSource* audio_source = nullptr;
+
+	GameObject* volume_settings = nullptr;
+	GameObject* resolution_settings = nullptr;
+
+	GameObject* gamepad_icons = nullptr;
+	GameObject* pc_icons = nullptr;
+
+	std::vector <ComponentTransform2D*> cursor_positions;
```
Comment (RickyVimon) [inline Scripts/src/Script/OptionsMenuController.h:72]: Delete this vector of pointers on the destructor

Comment (Pacasasgar) [reply Scripts/src/Script/OptionsMenuController.h:72]: As commented with Mario, this is not needed.

File: Scripts/src/Script/OptionsMenuController.h (hunk 2 above)
Comment (Pacasasgar) [reply Scripts/src/Script/OptionsMenuController.h:42]: Done in the default destructor

File: Scripts/src/Script/OptionsMenuController.cpp (hunk 1 above)
Comment (Pacasasgar) [reply Scripts/src/Script/OptionsMenuController.cpp:23]: Done!

**Review by RickyVimon — APPROVED** [review]
LGTM GJ Man

**Review by Dartemiss — APPROVED** [review]

**Pull request merged**